core/src/main/java/org/apache/cxf/attachment/LazyDataSource.java

```diff
@@ -47,7 +50,13 @@ public LazyDataSource(String id, Collection<Attachment> attachments) {
     private synchronized void load() {
         if (dataSource == null) {
             for (Attachment a : attachments) {
-                if (a.getId().equals(id)) {
+                String attachmentId = a.getId();
```

Comment:
@atnyxp thanks for highlighting the issue, I think we should not pursue this route and fix the encoding issue (so `attachmentId` would match `id`), I will be looking into it shortly

Reply:
It seems that this change https://github.com/apache/cxf/pull/1028 should fix it as well. Given the `attachmentId` which is valid domain, it should have no URL encode/decode issue in equality checks

Reply:
@reta Did you ever look into this again?

Reply:
@coheigea I suspect this change is not needed anymore, we've altered the way encode/decode is done

Reply:
Thanks @reta , let's close the PR